hera_qm/xrfi.py

```diff
@@ -1639,6 +1658,310 @@ def xrfi_run(ocalfits_file, acalfits_file, model_file, data_file, history,
+    """Run the xrfi excision pipeline used for H3C IDR2.1.
+
+    This pipeline uses the detrending and watershed algorithms above.
+    Several files are concatenated together to perform the detrending,
+    and the flags from the inner files are stored.
+    The algorithm is run on several data products: omnical gains, omnical chisq,
+    abscal gains, abscal chisq, omnical visibility solutions, renormalized chisq,
+    and the raw data. All of these, except the data, are run twice - first to
+    get an initial estimate of heavily contaminated data, and a second time
+    to get better estimate. The metrics and flags from each data product and both
+    rounds are stored in the xrfi_path (which defaults to a subdirectory, see
+    xrfi_path below). Also stored are the a priori flags and combined metrics/flags.
+
+    Parameters
+    ----------
+    ocalfits_files : str
+        The omnical calfits files to use to flag on gains and chisquared values.
+    acalfits_files : str
+        The abscal calfits files to use to flag on gains and chisquared values.
+    model_files : str
+        THe model visibility files to flag on.
+    data_files : str
+        The raw visibility data files to flag.
+    history : str
+        The history string to include in files.
+    xrfi_path : str, optional
+        Path to save xrfi files to. Default is a subdirectory "{JD}/" inside
+        the same directory as data_file.
+    kt_size : int, optional
+        The size of kernel in time dimension for detrend in xrfi algorithm.
+        Default is 8.
+    kf_size : int, optional
+        Size of kernel in frequency dimension for detrend in xrfi algorithm.
+        Default is 8.
+    sig_init : float, optional
+        The starting number of sigmas to flag on. Default is 5.0.
+    sig_adj : float, optional
+        The number of sigmas to flag on for data adjacent to a flag.
+        Default is 2.0.
+    ex_ants : str, optional
+        A comma-separated list of antennas to exclude. Flags of visibilities formed
+        with these antennas will be set to True. Default is None (i.e., no antennas
+        will be excluded).
+    metrics_file : str, optional
+        Metrics file that contains a list of excluded antennas. Flags of visibilities
+        formed with these antennas will be set to True. Default is None (i.e.,
+        no antennas will be excluded).
+    clobber : bool, optional
+        If True, overwrite existing files. Default is False.
+    run_check : bool
+        Option to check for the existence and proper shapes of parameters
+        on UVFlag Object.
+    check_extra : bool
+        Option to check optional parameters as well as required ones.
+    run_check_acceptability : bool
+        Option to check acceptable range of the values of parameters
+        on UVFlag Object.
+
+    Returns
+    -------
+    None
+
+    """
+    history = 'Flagging command: "' + history + '", Using ' + hera_qm_version_str
```

Comment:
It's slightly weird that whatever you put in the "history" is labeled as a flagging command, even if that's how it's primarily used. Maybe change the argument name?